### ISSUE TITLE
Proposal: List all files involved in DuplicateModule error

### ISSUE THREAD

**Issue by milesfrain** [purescript/purescript#3884]
It would be more helpful to list all duplicate files detected, rather than just one of them.

For example:
```
spago init
mkdir src/sneaky
echo "module Main where" > src/sneaky/Dup.purs
spago build
```
```
Error found:
at src/Main.purs:1:1 - 6:28 (line 1, column 1 - line 6, column 28)

  Module Main has been defined multiple times

See https://github.com/purescript/documentation/blob/master/errors/DuplicateModule.md for more information,
or to contribute content related to this error.

[error] Failed to build.
```
I'm actually more interested in knowing about `src/sneaky/Dup.purs` than `src/Main.purs`.

Related to #1828

**Comment by hdgarrood** [comment]
It looks like this previously worked. The fact that it now doesn’t may be related to #3273. It seems like there is a mechanism for including more than one source span in an error, except that in the case of this error we’re only actually printing one of the spans, whereas we should really be printing all of them?

**Comment by garyb** [comment]
Yeah, it might be that we never print the multiple spans when they're available, and only take the primary/first one. 

I was also a little disappointed that none of the tools make use of the multi-spans that I know of after they got added, but then it's not like I did anything to resolve that either. :wink:
